mobile/src/main/java/rs/readahead/washington/mobile/views/activity/viewer/VideoViewerActivity.kt

```diff
@@ -306,39 +295,32 @@ class VideoViewerActivity : BaseLockActivity(), StyledPlayerControlView.Visibili
     private fun initializePlayer() {
         val needNewPlayer = player == null
         if (needNewPlayer) {
-            player = ExoPlayer.Builder(this).build()
-            player?.apply {
-                binding.playerView.player
+            player = ExoPlayer.Builder(this).build().apply {
                 playWhenReady = shouldAutoPlay
             }
-
-            simpleExoPlayerView.setPlayer(player)
+            simpleExoPlayerView.player = player
         }
+
         if (needNewPlayer || needRetrySource) {
-            if (intent.hasExtra(VIEW_VIDEO) && intent.extras != null) {
-                val vaultFile = intent.extras!![VIEW_VIDEO] as VaultFile?
-                if (vaultFile != null) {
-                    this.vaultFile = vaultFile
-                    toolbar.title = vaultFile.name
-                    setupMetadataMenuItem(vaultFile.metadata != null)
-                    val mediaFileDataSourceFactory = MediaFileDataSourceFactory(
-                        this, vaultFile, null
-                    )
-                    val mediaItem =
-                        MediaItem.fromUri(MediaFileHandler.getEncryptedUri(this, vaultFile))
-                    val mediaSource: MediaSource =
-                        ProgressiveMediaSource.Factory(mediaFileDataSourceFactory)
-                            .createMediaSource(mediaItem)
-
-                    val haveResumePosition = resumeWindow != C.INDEX_UNSET
-                    if (player != null) {
-                        if (haveResumePosition) {
-                            player?.seekTo(resumeWindow, resumePosition)
-                        }
-                        player?.prepare(mediaSource, !haveResumePosition, false)
+            val vaultFile = intent.getSerializableExtra(VIEW_VIDEO) as? VaultFile
```

Comment:
maybe we should apply safe casting with the elvis operator here **?:**

`    val vaultFile = intent.getSerializableExtra(VIEW_VIDEO) as? VaultFile ?: return
`

**return statement is used to exit the function early, avoiding any further execution of the code.**